### PR TITLE
Migrate UserObjects to new warehouse

### PR DESCRIPTION
**Pull request by aeslaughter**
see #6375 
(refs #5680)

### CODE REVIEW

**Comment by aeslaughter** [comment]
```
$ git diff upstream/devel --shortstat -- framework/
 39 files changed, 571 insertions(+), 1614 deletions(-)
```

File: framework/src/userobject/GeneralUserObject.C [hunk 1]
```diff
@@ -73,3 +73,15 @@ GeneralUserObject::getVectorPostprocessorValueByName(const VectorPostprocessorNa
   _depend_vars.insert(name);
   return VectorPostprocessorInterface::getVectorPostprocessorValueByName(name, vector_name);
 }
+
+void
+GeneralUserObject::threadJoin(const UserObject &)
+{
+  mooseError("GeneralUserObjects do not execute using threads, this function does nothing and should not be used.");
```
Comment (permcody) [inline framework/src/userobject/GeneralUserObject.C]: This is good!

**Comment by permcody** [comment]
This looks good! :+1:

**Comment by YaqiWang** [comment]
Wait, Yak is failing with this PR. Right?

**Comment by permcody** [comment]
It is, MOOSE will not merge to master unless you merge. You have so much
power right now!

**Comment by dschwen** [comment]
Similar power dynamics as a hostage situation...

**Comment by permcody** [comment]
Yes but I'm confused about who the hostage is in this situation. Is it Yaqi
for being forced to merge a bunch of code, or the MOOSE Team for not being
able to merge to master if Yaqi doesn't merge the code?

eh... "master" branch? I never used it anyway...

**Comment by YaqiWang** [comment]
Either way Yak is the hostage, just who is the one kidnapping Yak...

**Pull request merged**